### DISCUSSION TITLE
A possible bug in  DAJacCondFdW.C

### DISCUSSION THREAD

**Discussion by xiaopingqiu** [mdolab/dafoam#73] [Bug] [answered]
Hello Dr He,
    I think one of my team member, Dr Ma,  have found a bug in DAJacCondFdW.C, function `setupJacCon`: 
```
label glbRowI = this->getGlobalObjFuncGeoIndex("cell", cellI);
```
In this line, `cellI` is the cell index, but it should be the index in the list `objFuncCellSources`, i.e., this line should be 
```
label glbRowI = this->getGlobalObjFuncGeoIndex("cell", idxI);
```

If the objective function is computed from cell values, but not all cells are involved, this bug causes a segmentation fault when constructing PETSC matrix. 

I notice that the only tutorial case that using objective function that is computed from cell value is `PlateHole_Structure`, but in this case, all cells are involved, the cell index will not exceed the size of list `objFuncCellSources`, so this case do not exit due to segmentation fault, but the result should not be right, in principle.

**Comment by friedenhe** [comment, marked as answer]
Thanks for the feedback. It has been fixed in this [commit](https://github.com/mdolab/dafoam/commit/34aab89cc7662a366ce1ffe6a3da31b342eb610f). Note that the PlateHole_Structure tutorial was not affected because all the cells are selected, so `idxI` and `cellI` are essentially the same.
